src/dac.rs

```diff
@@ -0,0 +1,355 @@
+            ($($output_ch1,)* Dac1Ch2<M_EXT_PIN, Disabled>)
+        );
+        impl_pin_for_dac!(DAC1: ($($pin_ch1,)* Dac1IntSig2),                    // ch2: Internal
+            ($($output_ch1,)* Dac1Ch2<M_INT_SIG, Disabled>)
+        );
+        impl_pin_for_dac!(DAC1: ($($pin_ch1,)* (PA5<DefaultMode>, Dac1IntSig2)),// ch2: Mixed
+            ($($output_ch1,)* Dac1Ch2<M_MIX_SIG, Disabled>)
+        );
+    };
+}
+
+impl_dac1_ch2_combos!(); // ch1: Not used
+impl_dac1_ch2_combos!(PA4<DefaultMode>, Dac1Ch1<M_EXT_PIN, Disabled>); // ch1: Ext pin
+impl_dac1_ch2_combos!(Dac1IntSig1, Dac1Ch1<M_INT_SIG, Disabled>); // ch1: Internal
+impl_dac1_ch2_combos!((PA4<DefaultMode>, Dac1IntSig1), Dac1Ch1<M_MIX_SIG, Disabled>); // ch1: Mixed
+
+// DAC2
+impl_pin_for_dac!(DAC2: PA6<DefaultMode>, Dac2Ch1<M_EXT_PIN, Disabled>); // ch1: Ext pin
+impl_pin_for_dac!(DAC2: Dac2IntSig1, Dac2Ch1<M_INT_SIG, Disabled>); // ch1: Internal
+impl_pin_for_dac!(DAC2: (PA6<DefaultMode>, Dac2IntSig1), Dac2Ch1<M_MIX_SIG, Disabled>); // ch1: Mixed
+
+// DAC3 int
+impl_pin_for_dac!(DAC3: Dac3IntSig1, Dac3Ch1<M_INT_SIG, Disabled>);
+impl_pin_for_dac!(DAC3: Dac3IntSig2, Dac3Ch2<M_INT_SIG, Disabled>);
+
+// DAC4 int
+impl_pin_for_dac!(DAC4: Dac4IntSig1, Dac4Ch1<M_INT_SIG, Disabled>);
+impl_pin_for_dac!(DAC4: Dac4IntSig2, Dac4Ch2<M_INT_SIG, Disabled>);
+
+pub fn dac<DAC, PINS>(_dac: DAC, _pins: PINS, _rcc: &mut Rcc) -> PINS::Output
+where
+    DAC: rcc::Enable + rcc::Reset,
+    PINS: Pins<DAC>,
+{
+    unsafe {
+        let rcc_ptr = &(*RCC::ptr());
+        DAC::enable(rcc_ptr);
+        DAC::reset(rcc_ptr);
+    }
+
+    #[allow(clippy::uninit_assumed_init)]
+    unsafe {
+        MaybeUninit::uninit().assume_init()
+    }
+}
+
+macro_rules! dac_helper {
+    ($($CX:ident: $DAC:ty: (
+        $en:ident,
+        $cen:ident,
+        $cal_flag:ident,
+        $trim:ident,
+        $mode:ident,
+        $dhrx:ident,
+        $dac_dor:ident,
+        $daccxdhr:ident,
+        $wave:ident,
+        $mamp:ident,
+        $ten:ident,
+        $swtrig:ident
+    ),)+) => {
+        $(
+            impl<const MODE_BITS: u8> $CX<MODE_BITS, Disabled> {
+                pub fn enable(self) -> $CX<MODE_BITS, Enabled> {
+                    let dac = unsafe { &(*<$DAC>::ptr()) };
+
+                    dac.dac_mcr.modify(|_, w| unsafe { w.$mode().bits(MODE_BITS) });
+                    dac.dac_cr.modify(|_, w| w.$en().set_bit());
+
+                    $CX {
+                        _enabled: PhantomData,
+                    }
+                }
+
+                /*pub fn enable_unbuffered(self) -> $CX<MODE_BITS, EnabledUnbuffered> {
+                    let dac = unsafe { &(*<$DAC>::ptr()) };
+
+                    dac.dac_mcr.modify(|_, w| unsafe { w.$mode().bits(2) });
+                    dac.dac_cr.modify(|_, w| w.$en().set_bit());
+
+                    $CX {
+                        _enabled: PhantomData,
+                    }
+                }*/
```

Comment:
Buffered vs unbuffered is pretty much chosen by MODE_BITS. Not sure how we want to handle that

Reply:
Use input argument as selector between modes